geode-serialization/src/main/java/org/apache/geode/internal/serialization/DSFIDSerializerImpl.java

```diff
@@ -111,19 +117,34 @@ public void writeDSFIDHeader(int dsfid, DataOutput out) throws IOException {
   }
 
   @Override
-  public void writeDSFID(DataSerializableFixedID o, DataOutput out) throws IOException {
-    if (o == null) {
-      out.writeByte(DSCODE.NULL.toByte());
+  public void write(final BasicSerializable bs, final DataOutput out) throws IOException {
+    if (writeMetaData(bs, out)) {
```

Comment:
I think this code might be better if you changed writeMetaData to return void and just had it write the meta data.
Then after you call it you could conditionalize the invokeToData call with "if (bs != null)".

Reply:
yes! changing…